config/v1/types_infrastructure.go

```diff
@@ -562,6 +616,14 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancer defines how the load balancer used by the cluster is configured.
+	// +default={"openShiftManaged": {"type": "Default"}, "type": "OpenShiftManaged"}
+	// +kubebuilder:default={openShiftManaged: {type: Default}, type: OpenShiftManaged}
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="loadBalancer is immutable once set"
```

Comment:
Curious whether this validation actually works on a struct, as you are comparing oldSelf to an empty string 🤔 You might have to add a validation to parent struct to check if the struct has the loadBalancer field in oldSelf and then compare them for equality